### PR TITLE
Dedup code.

### PR DESCRIPTION
**Pull request by wujingyue**

### CODE REVIEW

File: csrc/fusion_segmenter.h [hunk 1]
```diff
@@ -524,13 +524,7 @@ class SegmentCandidateFinder {
...
       SegmentCandidateFinderOptions options = SegmentCandidateFinderOptions()) {
     auto fusion_copy = std::make_unique<Fusion>(*fusion);
-    if (isDebugDumpEnabled(DebugDumpOption::FusionSegments)) {
-      debug() << "Segment the fusion (Original Fusion Un-modified): "
-              << std::endl;
-      fusion_copy->printMath();
-    }
-    SegmentCandidateFinder scf(std::move(fusion_copy), inputs, options);
-    return std::move(scf.segmented_fusion_);
+    return segment(std::move(fusion_copy), inputs, options);
```
Comment (naoyam) [inline csrc/fusion_segmenter.h:527]: The change looks good, but I think the debug dump in the other version of `segment` is incorrectly placed after segmentation. At line 535, the complete fusion is segmented, during which some additional ops may be inserted, so `scf.completeFusion()` may be modified. I think it's supposed to be done before line 535.

Comment (naoyam) [reply csrc/fusion_segmenter.h:527]: @wujingyue Could you also fix the print location?

Comment (wujingyue) [reply csrc/fusion_segmenter.h:527]: Done.

**Review by naoyam — APPROVED** [review]
LGTM

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
@zasdfgbnm The 11 TMATest failures for H100 seem to be irrelevant to this PR. But raising it FYI

**Pull request merged**